### PR TITLE
[Rust] Fixed a bug related to wasi module initialization

### LINKED ISSUE #1580: No wasi env has been passed with wasmedge_sys
My wasm(rust):
```rust
use std::env;

#[no_mangle]
pub fn print_env() {
  println!("The env vars are as follows.");
  for (key, value) in env::vars() {
    println!("{}: {}", key, value);
  }

  println!("The args are as follows.");
  for argument in env::args() {
    println!("{}", argument);
  }
}
```

When I call `print_env` with wasmedge binary:
`wasmedge --env a=b --env c=d --reactor test.wasm print_env`

The result is ok:
```
The env vars are as follows.
a: b
c: d
The args are as follows.
test.wasm
print_env
```

But if I call `print_env` using wasmedge_sys:
```rust
use wasmedge_sys::{Config, Vm};
use std::path::Path;

fn main() {
    let mut config = Config::create().unwrap();
    config.wasi(true);

    let mut vm = Vm::create(Some(config), None).unwrap();

    // get default wasi module
    let mut wasi_module = vm.wasi_module_mut().unwrap();
    // init the default wasi module
    wasi_module.init_wasi(
        Some(vec!["abc", "123"]),
        Some(vec!["a=b"]),
        Some(vec![]),
    );  

    let wasm_path = Path::new("/home/test.wasm");
    let _ = vm.load_wasm_from_file(wasm_path);
    let _ = vm.validate();
    let _ = vm.instantiate();

    println!("{:?}", vm.run_function("print_env", vec![]));
}
```
I got:
```
The env vars are as follows.
The args are as follows.


Ok([])
```

Envs has not been passed in, and the two empty output lines showed that the two args had been passed in without content.

### PR DESCRIPTION
**Pull request by apepkuss**
This PR fixed #1580.

### CODE REVIEW

**Comment by apepkuss** [comment]
Hi @hydai, could you please help review this PR? Thanks a lot!

**Comment by hydai** [comment]
The failure on macos is related to #1610.

File: bindings/rust/wasmedge-sys/src/instance/module.rs [hunk 1]
```diff
@@ -543,35 +542,71 @@ impl WasiModule {
...
+        // let envs_len = envs.len();
+
+        // let preopens = match preopens {
+        //     Some(preopens) => preopens
+        //         .into_iter()
+        //         .map(string_to_c_char)
+        //         .collect::<Vec<_>>(),
+        //     None => vec![],
+        // };
+        // let preopens_len = preopens.len();
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/module.rs:600]: Should we remove these comment?

File: bindings/rust/wasmedge-sys/src/instance/module.rs [hunk 2]
```diff
@@ -919,16 +969,28 @@ impl WasmEdgeProcessModule {
...
-        let cmds_len = cmds.len();
+        let mut p_cmds: Vec<_> = cstr_cmds.iter().map(|x| x.as_ptr()).collect();
+        let p_cmds_len = p_cmds.len();
+        p_cmds.push(std::ptr::null());
+
+        // let cmds = match allowed_cmds {
+        //     Some(cmds) => cmds.iter().map(string_to_c_char).collect::<Vec<_>>(),
+        //     None => vec![],
+        // };
+        // let cmds_len = cmds.len();
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/module.rs:988]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/examples/wasi_print_env.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    let mut config = Config::create()?;
+    config.wasi(true);
+
+    // create a vm
+    let mut vm = Vm::create(Some(config), None)?;
+
+    // set the envs and args for the wasi module
+    let args = vec!["arg1", "arg2"];
+    let envs = vec!["ENV1=VAL1", "ENV2=VAL2", "ENV3=VAL3"];
+    // let envs = vec!["hello=world", "ENV2=VAL2"];
```
[the remark this answers is not in the document]
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/examples/wasi_print_env.rs:16]: Accepted. Thanks!

File: bindings/rust/wasmedge-sys/src/instance/module.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/module.rs:600]: Accepted. Thanks!

File: bindings/rust/wasmedge-sys/src/instance/module.rs (hunk 2 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/module.rs:988]: Accepted. Thanks!

**Comment by hydai** [comment]
Since I merged the #1602 first, please resolve the conflicts.

**Comment by apepkuss** [comment]
> Since I merged the #1602 first, please resolve the conflicts.

Rebased on the latest master branch. Thanks!

**Review by hydai — APPROVED** [review]

**Comment by apepkuss** [comment]
@hydai Thanks for the review!

**Pull request merged**